pcsx2/gui/GlobalCommands.cpp

```diff
@@ -150,15 +150,20 @@ namespace Implementations
 			OSDlog( Color_StrongRed, true, "(FrameLimiter) SlowMotion ENABLED." );
 			g_Conf->EmuOptions.GS.FrameLimitEnable = true;
 		}
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
+
 		pauser.AllowResume();
 	}
 
 	void Framelimiter_MasterToggle()
 	{
 		ScopedCoreThreadPause pauser;
 		g_Conf->EmuOptions.GS.FrameLimitEnable = !g_Conf->EmuOptions.GS.FrameLimitEnable;
-		GSsetVsync( g_Conf->EmuOptions.GS.FrameLimitEnable && g_Conf->EmuOptions.GS.VsyncEnable );
 		OSDlog( Color_StrongRed, true, "(FrameLimiter) %s.", g_Conf->EmuOptions.GS.FrameLimitEnable ? "ENABLED" : "DISABLED" );
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
```

Comment:
Conditionally decouple framelimiter from vsync only for certain refresh rates sounds really bad. 
(also, with variable refresh rates monitors, situations might \*_really_\* become awkward)

Btw, I think "Buffer Object Streaming" is the better naming/technique for the buffering thing I had in my mind yesterday. 
Which certainly wouldn't lead us back to square one if what I'm supposing in #1887 is right.